rules/TypeDeclaration/Rector/Expression/InlineVarDocTagToAssertRector.php

```diff
@@ -0,0 +1,130 @@
+use PHPStan\PhpDocParser\Ast\PhpDoc\VarTagValueNode;
+use Rector\BetterPhpDocParser\PhpDocInfo\PhpDocInfo;
+use Rector\BetterPhpDocParser\PhpDocInfo\PhpDocInfoFactory;
+use Rector\BetterPhpDocParser\PhpDocManipulator\PhpDocTagRemover;
+use Rector\Comments\NodeDocBlock\DocBlockUpdater;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use Rector\Rector\AbstractRector;
+use Rector\TypeDeclaration\PhpDocParser\TypeExpressionFromVarTagResolver;
+use Rector\ValueObject\PhpVersionFeature;
+use Rector\VersionBonding\Contract\MinPhpVersionInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\TypeDeclaration\Rector\Expression\InlineVarDocTagToAssertRector\InlineVarDocTagToAssertRectorTest
+ */
+final class InlineVarDocTagToAssertRector extends AbstractRector implements MinPhpVersionInterface
+{
+    public function __construct(
+        private readonly PhpDocInfoFactory $phpDocInfoFactory,
+        private readonly PhpDocTagRemover $phpDocTagRemover,
+        private readonly DocBlockUpdater $docBlockUpdater,
+        private readonly TypeExpressionFromVarTagResolver $typeExpressionFromVarTagResolver
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Convert inline @var tags to calls to assert()', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+/** @var Foo $foo */
+$foo = createFoo();
+CODE_SAMPLE
+
+                ,
+                <<<'CODE_SAMPLE'
+$foo = createFoo();
+assert($foo instanceof Foo);
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Expression::class];
+    }
+
+    /**
+     * @param Expression $node
+     * @return Node[]|null
+     */
+    public function refactor(Node $node): ?array
+    {
+        if (! $node->expr instanceof Assign) {
+            return null;
+        }
+
+        if (! $node->expr->var instanceof Variable) {
+            return null;
+        }
+
+        $docComment = $node->getDocComment();
+        if (! $docComment instanceof Doc) {
+            return null;
+        }
+
+        $phpDocInfo = $this->phpDocInfoFactory->createFromNode($node);
+
+        if (! $phpDocInfo instanceof PhpDocInfo || $phpDocInfo->getPhpDocNode()->children === []) {
+            return null;
+        }
+
+        $expressionVariableName = $node->expr->var->name;
+        foreach ($phpDocInfo->getPhpDocNode()->children as $phpDocChildNode) {
+            if (! $phpDocChildNode instanceof PhpDocTagNode) {
+                continue;
+            }
+
+            $tagValueNode = $phpDocChildNode->value;
+            // handle only basic types, keep phpstan/psalm helper ones
+            if ($tagValueNode instanceof VarTagValueNode && $phpDocChildNode->name === '@var') {
```

Comment:
to exclude `phsptan-` or `psalm-` prefix, you can remove fro `$phpDocInfo` object so no need manually check it.

https://github.com/rectorphp/rector-src/blob/71fb2e4c9346bf6ce6b057ff3d77511662f7913c/src/BetterPhpDocParser/PhpDocInfo/PhpDocInfo.php#L256-L274

to be:

```php
$phpDocInfo->removeByType(VarTagValueNode::class);
$this->docBlockUpdater->updateRefactoredNodeWithPhpDocInfo($node);
```

please add test for skip this as well.

Reply:
I cannot use `$phpDocInfo->removeByType(VarTagValueNode::class);` because that would remove other @var tags, not just the one that affects our current variable. I added some new tests to check the psalm and phpstan versions

Reply:
You need to pass 2nd arg as variable name:

```php
$phpDocInfo->removeByType(VarTagValueNode::class, $variableName);
```

Reply:
Ah, yes, you are right, updated